File: Napakalaki Ruby/Treasure.rb
31# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame

class Treasure
  
  private
  attr_reader :name, :bonus, :type
  
  public
  def initialize(n,bono,t) #String n, int bonus, TreasureKind t
    @name = n
    @bonus = bono
    @type = t
    
  end
  
  def to_s()
    cadena = "Nombre: #{@name}\n"+
             "Bonus: #{@bonus}\n"+
             "Tipo: #{@type}\n"
    return cadena
  end

    
  def getName 
    return @name
  end
  
  def getBonus
    return @bonus
  end
  
  def getType
    return @type
  end
  
  
end

end

File: Napakalaki Java/NapakalakiGame/Treasure.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

/**
 *
 * @author Germán
 */
public class Treasure {
    private String name;
    private int bonus;
    private TreasureKind treasure;
    
    
    public Treasure(String n, int b, TreasureKind t) {
        name = n;
        bonus = b;
        treasure = t;
    }
    
    public String toString() {
        return "Nombre del tesoro: " + name + "\nBonus del tesoro: " 
                + Integer.toString(bonus)
                + "\nTipo de tesoro: " + treasure; 
        
    }
    
    public String getName () {
        
        return name;
    }
    
    public int getBonus () {
        
        return bonus;
    }
    
    public TreasureKind getType () {
        
        return treasure;
    }
}
